### ISSUE TITLE
Slow validation

### ISSUE THREAD

**Issue by rawsh** [PRIME-RL/PRIME#20]
Hello! Trying out PRIME with a 0.5B model on a single H100 and seeing very slow validation - "validation generation end" was printed ~250 times in about two hours. 

From what I can tell it looks the valset is split into batches and this is printed after every batch. I'm using the example settings with the `PRIME-RL/Eurus-2-RL-Data` validation set and `data.val_batch_size=4`. I would expect ~500 batches to run much faster with a 0.5B model + H100, is there something I'm missing? Is there any way to speed up the training?

From other comments I've set `data.filter_accuracy=False` and set all offload parameters to `False`. Should I be limiting the output sequence length aggressively? How can this be set? Thanks!

**Comment by cgq15** [comment]
I think a large batch size would be faster. For 0.5B model, I suggest setting val_batch_size to 1024. 

`data.filter_accuracy` is for training prompt filtering and does not affect memory. You can try different sequence lengths. For 7B models, it is ok to set 8k length on a single node of A100s.

**Comment by cgq15** [comment]
Actually, filtering would slow down training since we discard some prompts that are too easy or too hard. but it stablizes training.

With a smaller model, I think you can use a larger batchsize to better ultilize your GPU memory.

**Comment by rawsh** [comment]
Gotcha, makes sense
One thing I'm confused on is the 0.5B model seems to be taking a similar amount of time as the 7B parameter model (I think I saw another comment Eurus-PRIME was trained in 3 days on 8xA100 80b?)

On 8xH100s using train batch size = 512, eval batch size = 1024, with offloading and accuracy filtering disabled, I'm seeing step times around 4.5 mins. With 940 steps that means full training would be around 2.8 days? Does this seem right for such a smaller model?

```
56 gen: 100.0 seconds
57 verify: 18.0 seconds
58 collected 2048 / 2048 rollouts and each prompt has 4 responses
59 rollout batch size: 2048
60 ref: 7.7 seconds
61 reward: 62.9 seconds
62 adv: 23.9 seconds
63 update_actor: 53.9 seconds
64 step:6 - timing/gen:99.961 - timing/verify:18.029 - train_verify_score/code:0.081 - train_verify_score/math:0.081 - train_verify_score/all:0.081 - timing/ref:7.656 - reward_model/prm_loss:0.674 - reward_model/grad_norm:12.078 - reward_model/dpo_acc_before:0.521 - timing/reward_model:62.945 - train_reward/code:0.081 - train_reward/math:0.081 - train_reward/all:0.081 - train_reward/verifier:0.081 - train_reward/reward_all:0.404 - critic/kl:0.000 - critic/kl_coeff:0.000 - timing/adv:23.938 - timing/update_actor:53.903 - actor/pg_loss:-0.796 - actor/pg_clipfrac:0.001 - actor/ppo_kl:0.000 - actor/grad_norm:0.504 - actor/lr(1e-4):0.005 - critic/score/mean:0.404 - critic/score/max:5.000 - critic/score/min:0.000 - critic/rewards/mean:0.404 - critic/rewards/max:5.000 - critic/rewards/min:0.000 - critic/advantages/mean:-0.000 - critic/advantages/max:4.522 - critic/advantages/min:-4.423 - critic/returns/mean:-0.011 - critic/returns/max:1.000 - critic/returns/min:-1.000 - response_length/mean:1588.778 - response_length/max:3072.000 - response_length/min:42.000 - prompt_length/mean:219.150 - prompt_length/max:875.000 - prompt_length/min:136.000
```

**Comment by cgq15** [comment]
With 0.5B model I think you can use a larger batchsize for actor and rm update, i.e. `mini_batch_size` and `micro_batch_size`.

Eurus-PRIME was trained with accuracy filtering enabled, which discarded many prompts (about 2/3), so the model was only trained for around 320 steps. A full training on these prompts would require longer time.

**Comment by rawsh** [comment]
Really appreciate the replies @cgq15 - Still a bit confused how accuracy filtering only effectively trains for 1/3 of the steps? Just from my testing it seemed like the main difference was it took multiple iterations to get a full rollout which slowed down the time for each step by around ~4-5x

**Comment by cgq15** [comment]
To breakdown, accuracy filter need more rollouts, so it makes **each step** slower. However, since many prompts are discarded after rollout, they are not used to update the PRM and actor. Therefore, the **overall epoch time** is reduced
